sdk/cosmos/azure-cosmos/azure/cosmos/_retry_utility.py

```diff
@@ -112,6 +133,31 @@ def Execute(client, global_endpoint_manager, function, *args, **kwargs):
                 retry_policy = sessionRetry_policy
             elif exceptions._partition_range_is_gone(e):
                 retry_policy = partition_key_range_gone_retry_policy
+            elif exceptions._container_recreate_exception(e):
+                retry_policy = container_recreate_retry_policy
+                # Before we retry if retry policy is container recreate, we need refresh the cache of the
+                # container properties and pass in the new RID in the headers.
+                client._refresh_container_properties_cache(retry_policy.container_link)
```

Comment:
This looks like it's making the refresh request before we've determined whether a retry should be made? (i.e. shouldn't this logic come after we've confirmed that `ShouldRetry == true`?)

Reply:
The refresh should happen before we retry and even if shouldretry were to return false. There are some edge cases where we might have a container recreate but we did not get RID_MISMATCH error which would require to compare the RID mismatch on the client side to determine a retry.